### PR TITLE
Some UI Adittions and search bar (🙀🙀🙀)

### PR DESCRIPTION
**Pull request by Ultradragon005**
Made Presetoption a manual thing because It's more convenient to edit and see it on the front page.

~~and this basically works now, Just need to fix some bugs~~

Fully works now except one very minor bug (only happens the first time u open the menu idk)

### CODE REVIEW

**Comment by Ultradragon005** [comment]
Can't wait til 💀😂
image

**Comment by HyperAtill** [comment]
> Can't wait til 💀😂
> image
> 

+

**Comment by Ultradragon005** [comment]
Ok it's done now, awaiting 2.1.0

File: Patches/ChatBubblePatch.cs [hunk 1]
```diff
@@ -28,4 +29,24 @@ public static void Postfix(ChatBubble __instance, [HarmonyArgument(1)] bool isDe
...
+    public static bool Prefix(TextMeshPro self, ref Vector2 __result)
+    {
+        if (self.textInfo == null || self.textInfo.lineCount == 0 || self.textInfo.lineInfo[0].characterCount <= 0)
+        {
+            __result = self.GetTextInfo(" ").lineInfo.First().lineExtents.max;
+            return false;
+        }
+
+        return true;
+    }
```
Comment (Ultradragon005) [inline Patches/ChatBubblePatch.cs:51]: gonna move this to textboxpatch

**Comment by Ultradragon005** [comment]
image

**Pull request merged**